test/export/InstanceExporter.test.ts

```diff
@@ -5886,6 +5886,30 @@ describe('InstanceExporter', () => {
       ]);
     });
 
+    it('should assign a code with a version to a top level element while replacing the code system name with its url when the correct version is found', () => {
+      const brightInstance = new Instance('BrightObservation');
+      brightInstance.instanceOf = 'Observation';
+      const assignedCodeRule = new AssignmentRule('code');
+      assignedCodeRule.value = new FshCode('bright', 'Visible|1.0.0');
+      brightInstance.rules.push(assignedCodeRule);
+      doc.instances.set(brightInstance.name, brightInstance);
+
+      const visibleSystem = new FshCodeSystem('Visible');
+      const visibleSystemVersion = new CaretValueRule('');
+      visibleSystemVersion.caretPath = 'url';
+      visibleSystemVersion.value = 'http://hl7.org/fhir/us/minimal/CodeSystem/Visible|1.0.0';
```

Comment:
I don't think a `CodeSystem` definition would declare its `url` with a version. The `url` would be the version-less URL and the `version` element would contain the version. E.g.
```ts
const visibleSystemUrl = new CaretValueRule('');
visibleSystemUrl.caretPath = 'url';
visibleSystemUrl.value = 'http://hl7.org/fhir/us/minimal/CodeSystem/Visible';
const visibleSystemVersion = new CaretValueRule('');
visibleSystemVersion.caretPath = 'version';
visibleSystemVersion.value = '1.0.0';
```

Reply:
I thought I had added this case because of a regression, but I'm not sure which repo without rerunning the regression. I believe this was also tied to [this](https://github.com/FHIR/sushi/pull/1275#discussion_r1196899479) change, and updating to use the regex replace actually causes this test to fail. I guess it is worth keeping the test with the version caret value rule, so I'll update this!

Reply:
Included in [70e31fc](https://github.com/FHIR/sushi/pull/1275/commits/70e31fc1b4bdb5eaab6119f21d08e44e038c24db)

Reply:
Technically, `CodeSystem.url` is a `uri`, so you're not supposed to append a version to it. It even says:
> URL should not contain | or # - these characters make processing canonical references problematic

So if someone is making a CodeSystem with a URL like that in a real IG, we shouldn't support it anyway.

Reply:
Oh! Well maybe I just invented that test case. It seems so specific so I thought I must have pulled it from somewhere, but maybe not. Well, it is no longer supported now!